### ISSUE TITLE
Plugin hooks not executed in plugin priority order

### ISSUE THREAD

**Issue by tobdub** [vernemq/vernemq#1295]
### Environment
- VerneMQ Version: 1.8.0
- OS: docker alpine linux
- VerneMQ configuration (vernemq.conf) or the changes from the default
```
allow_anonymous = off

plugins.vmq_diversity = on
vmq_diversity.auth_postgres.enabled = on
vmq_diversity.postgres.host = db
vmq_diversity.postgres.port = 5432
vmq_diversity.postgres.user = vernemq
vmq_diversity.postgres.password = vernemq
vmq_diversity.postgres.database = vernemq
vmq_diversity.postgres.ssl = off

plugins.vmq_webhooks = on
vmq_webhooks.webhook_reg.hook = auth_on_register
vmq_webhooks.webhook_reg.endpoint = http://jwt-auth:5000
vmq_webhooks.webhook_sub.hook = auth_on_subscribe
vmq_webhooks.webhook_sub.endpoint = http://jwt-auth:5000
vmq_webhooks.webhook_pub.hook = auth_on_publish
vmq_webhooks.webhook_pub.endpoint = http://jwt-auth:5000

plugins.vmq_diversity.priority = 10

plugins.vmq_passwd = off
plugins.vmq_acl = off
plugins.vmq_bridge = off
```
- Cluster size/standalone: 1 node

### Expected behavior
The plugin hooks should be executed in the order specified by the plugin priority.

### Actual behaviour
The plugin hooks are executed out of order.
The output of `vmq-admin plugin show --internal` indicates that the plugin hooks are interleaved.
```
+------------------+----------+--------------------------+-----------------------------------------+
|      Plugin      |   Type   |         Hook(s)          |                  M:F/A                  |
+------------------+----------+--------------------------+-----------------------------------------+
|   vmq_plumtree   |applicatio|       metadata_put       |       vmq_plumtree:metadata_put/3       |
|                  |          |       metadata_get       |       vmq_plumtree:metadata_get/2       |
|                  |          |     metadata_delete      |     vmq_plumtree:metadata_delete/2      |
|                  |          |      metadata_fold       |      vmq_plumtree:metadata_fold/3       |
|                  |          |    metadata_subscribe    |    vmq_plumtree:metadata_subscribe/1    |
|                  |          |       cluster_join       |       vmq_plumtree:cluster_join/1       |
|                  |          |      cluster_leave       |      vmq_plumtree:cluster_leave/1       |
|                  |          |     cluster_members      |     vmq_plumtree:cluster_members/0      |
|                  |          |  cluster_rename_member   |  vmq_plumtree:cluster_rename_member/2   |
|                  |          |cluster_events_add_handler|vmq_plumtree:cluster_events_add_handler/2|
|                  |          |cluster_events_delete_hand|vmq_plumtree:cluster_events_delete_handle|
|                  |          |cluster_events_call_handle|vmq_plumtree:cluster_events_call_handler/|
|    vmq_config    |  module  |      change_config       |       vmq_config:change_config/1        |
| vmq_lvldb_store  |  module  |     msg_store_write      |    vmq_lvldb_store:msg_store_write/2    |
| vmq_lvldb_store  |  module  |     msg_store_delete     |   vmq_lvldb_store:msg_store_delete/2    |
| vmq_lvldb_store  |  module  |      msg_store_find      |    vmq_lvldb_store:msg_store_find/1     |
| vmq_lvldb_store  |  module  |      msg_store_read      |    vmq_lvldb_store:msg_store_read/2     |
|  vmq_diversity   |applicatio|                          |                                         |
|   vmq_webhooks   |applicatio|                          |                                         |
|vmq_diversity_plug|  module  |     auth_on_publish      | vmq_diversity_plugin:auth_on_publish/6  |
|vmq_webhooks_plugi|  module  |     auth_on_publish      |  vmq_webhooks_plugin:auth_on_publish/6  |
|vmq_diversity_plug|  module  |    auth_on_publish_m5    |vmq_diversity_plugin:auth_on_publish_m5/7|
|vmq_webhooks_plugi|  module  |     auth_on_register     | vmq_webhooks_plugin:auth_on_register/5  |
|vmq_diversity_plug|  module  |     auth_on_register     | vmq_diversity_plugin:auth_on_register/5 |
|vmq_webhooks_plugi|  module  |    auth_on_subscribe     | vmq_webhooks_plugin:auth_on_subscribe/3 |
|vmq_diversity_plug|  module  |   auth_on_register_m5    |vmq_diversity_plugin:auth_on_register_m5/|
|vmq_diversity_plug|  module  |    auth_on_subscribe     |vmq_diversity_plugin:auth_on_subscribe/3 |
|vmq_diversity_plug|  module  |   auth_on_subscribe_m5   |vmq_diversity_plugin:auth_on_subscribe_m5|
|vmq_diversity_plug|  module  |      on_client_gone      |  vmq_diversity_plugin:on_client_gone/1  |
|vmq_diversity_plug|  module  |    on_client_offline     |vmq_diversity_plugin:on_client_offline/1 |
|vmq_diversity_plug|  module  |      on_unsubscribe      |  vmq_diversity_plugin:on_unsubscribe/3  |
+------------------+----------+--------------------------+-----------------------------------------+
```
*Steps to reproduce*
Run with the plugin config above. Subscribe to a topic. The _vmq_diversity_ `auth_on_register` hook is expected to be called before the _vmq_webhooks_ `auth_on_register` hook.

**Comment by larshesel** [comment]
Thanks for opening the issue - we'll look into it as soon as possible.

**Comment by larshesel** [comment]
@dergraf, at the moment we're lazily registering the hooks as required by the scripts / hooks - I think we did it like that as we used to not have any insight in to the actual hooks used, but nowadays we have both `vmq-admin script show` and `vmq-admin webhooks show` which displays the active/configured hooks. I would therefore change the `vmq_webhooks` and `vmq_diversity` hooks to simply implement *all* available hooks and return sensible defaults/do nothing. Then the hook order would be exactly as the plugin order.

**Comment by larshesel** [comment]
Hi @tobdub, I just created a PR for this (it's not yet clear if this is the final fix), would you be able to test it out? See #1307.

**Comment by tobdub** [comment]
@larshesel Nice!
I did some quick testing and I can confirm that the fix seems to solve our problem. The plugin hooks are executed in the plugin priority order.

**Comment by larshesel** [comment]
Thanks for testing - we just merged it and expect to release 1.9.2 with the fix in a few days!
